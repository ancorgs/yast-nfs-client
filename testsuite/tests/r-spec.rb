# encoding: utf-8

# Module:
#   NFS client configuration
#
# Summary:
#   Routines testuite
#
# Authors:
#   Martin Vidner <mvidner@suse.cz>
#
# $Id$
module Yast
  class RSpecClient < Client
    def main
      # testedfiles: routines.ycp Testsuite.ycp

      Yast.include self, "testsuite.rb"
      Yast.include self, "nfs/routines.rb"

      DUMP("SpecToServPath")
      TEST(lambda { SpecToServPath("big.foo.com:/share/data") }, [], nil)
      TEST(lambda { SpecToServPath("only.server.com:") }, [], nil)
      TEST(lambda { SpecToServPath("nocolon.only.server.com") }, [], nil)
      TEST(lambda { SpecToServPath(":/only/path") }, [], nil)
      TEST(lambda { SpecToServPath("/nocolon/only/path") }, [], nil)
      TEST(lambda { SpecToServPath("fe80::219:d1ff:feac:fd10:/path") }, [], nil)
      TEST(lambda { SpecToServPath("") }, [], nil)

      nil
    end
  end
end

Yast::RSpecClient.new.main
