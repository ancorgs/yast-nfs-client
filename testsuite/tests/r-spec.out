Dump	SpecToServPath
Return	`couple ("big.foo.com", "/share/data")
Return	`couple ("only.server.com", "")
Return	`couple ("", "nocolon.only.server.com")
Return	`couple ("", "/only/path")
Return	`couple ("", "/nocolon/only/path")
Return	`couple ("two", "colons:used")
Return	`couple ("", "")
